pkg/machine/machineset.go

```diff
@@ -0,0 +1,378 @@
+	Object *machinev1beta1.MachineSet
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+	// errorMsg is processed before MachineSetBuilder object is created.
+	errorMsg string
+}
+
+// NewMachineSetBuilderFromCopy returns an MachineSetBuilder struct from a copied MachineSet.
+func NewMachineSetBuilderFromCopy(
+	apiClient *clients.Settings,
+	nsName,
+	publicCloud,
+	instanceType string,
+	replicas int32) *MachineSetBuilder {
+	glog.V(100).Infof(
+		"Initializing new MachineSetBuilder structure from copied MachineSet with the following params: "+
+			"namespace: %s, publicCloud: %s, instanceType: %s and replicas: %v",
+		nsName, publicCloud, instanceType, replicas)
+
+	newMachineSet, err := CreateNewWorkerMachineSetFromCopy(apiClient, nsName, publicCloud, instanceType, replicas)
+
+	builder := MachineSetBuilder{
+		apiClient:  apiClient,
+		Definition: newMachineSet,
+	}
+
+	if err != nil {
+		glog.V(100).Infof(
+			"Error initializing MachineSet from copy: %s", err.Error())
+
+		builder.errorMsg = fmt.Sprintf("Error initializing MachineSet from copy: %s",
+			err.Error())
+	}
+
+	if nsName == "" {
+		glog.V(100).Infof("The Namespace of the MachineSet is empty")
+
+		builder.errorMsg = "MachineSet 'nseName' cannot be empty"
+	}
+
+	if instanceType == "" {
+		glog.V(100).Infof("The instanceType of the MachineSet is empty")
+
+		builder.errorMsg = "MachineSet 'instanceType' cannot be empty"
+	}
+
+	if replicas == 0 {
+		glog.V(100).Infof("The replicas of the MachineSet is zero")
+
+		builder.errorMsg = "MachineSet 'replicas' cannot be zero"
+	}
+
+	if builder.Definition == nil {
+		glog.V(100).Infof("The MachineSet object definition is nil")
+
+		builder.errorMsg = "MachineSet 'Object.Definition' is nil"
+	}
+
+	return &builder
+}
+
+// PullMachineSet loads an existing MachineSet into Builder struct.
+func PullMachineSet(apiClient *clients.Settings, name, namespace string) (*MachineSetBuilder, error) {
+	glog.V(100).Infof("Pulling existing machineSet name %s in namespace %s", name, namespace)
+
+	builder := MachineSetBuilder{
+		apiClient: apiClient,
+		Definition: &machinev1beta1.MachineSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+		},
+	}
+
+	if name == "" {
+		builder.errorMsg = "MachineSet 'name' cannot be empty"
+	}
+
+	if namespace == "" {
+		builder.errorMsg = "MachineSet 'namespace' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("MachineSet object %s doesn't exist in namespace %s", name, namespace)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// CreateNewWorkerMachineSetFromCopy returns a MachineSet object which was copied from an exiting one on cluster.
+func CreateNewWorkerMachineSetFromCopy(
+	apiClient *clients.Settings,
+	namespace,
+	publicCloud,
+	instanceType string,
+	replicas int32) (*machinev1beta1.MachineSet, error) {
+	// currently only supporting AWS MachineSets
+	if publicCloud != "aws" {
```

Comment:
I recommend creating a private function that returns a slice of supported public clouds and checking publicCloud against it. Then as you add more supported public clouds you can just add them to the slice